compiler/src/parsing/parser.messages

```diff
@@ -3285,14 +3297,76 @@ program: LBRACE UIDENT COMMA UIDENT COMMA EOL WHILE
 ## may provide an INCOMPLETE view of the future (what was expected next).
 ## In state 3, spurious reduction of production nonempty_list(eol) -> EOL
 ## In state 5, spurious reduction of production eols -> nonempty_list(eol)
-## In state 96, spurious reduction of production comma -> COMMA eols
+## In state 94, spurious reduction of production comma -> COMMA eols
+##
+program: LBRACE ELLIPSIS BIGINT COMMA ELLIPSIS BIGINT ARROW
+##
+## Ends in an error in state: 227.
+##
+## lseparated_nonempty_list_inner(comma,record_field) -> lseparated_nonempty_list_inner(comma,record_field) . comma punned_record_field [ RBRACE EOL COMMA ]
+## lseparated_nonempty_list_inner(comma,record_field) -> lseparated_nonempty_list_inner(comma,record_field) . comma non_punned_record_field [ RBRACE EOL COMMA ]
+## lseparated_nonempty_list_inner(comma,record_field) -> lseparated_nonempty_list_inner(comma,record_field) . comma spread_record_field [ RBRACE EOL COMMA ]
+## record_exprs -> spread_record_field comma lseparated_nonempty_list_inner(comma,record_field) . option(comma) [ RBRACE EOL ]
+##
+## The known suffix of the stack is as follows:
+## spread_record_field comma lseparated_nonempty_list_inner(comma,record_field)
+##
+## WARNING: This example involves spurious reductions.
+## This implies that, although the LR(1) items shown above provide an
+## accurate view of the past (what has been recognized so far), they
+## may provide an INCOMPLETE view of the future (what was expected next).
+## In state 171, spurious reduction of production non_assign_expr -> left_accessor_expr
+## In state 125, spurious reduction of production non_binop_expr -> non_assign_expr
+## In state 69, spurious reduction of production annotated_expr -> non_binop_expr
+## In state 203, spurious reduction of production non_stmt_expr -> annotated_expr
+## In state 59, spurious reduction of production expr -> non_stmt_expr
+## In state 208, spurious reduction of production spread_record_field -> ELLIPSIS expr
+## In state 224, spurious reduction of production lseparated_nonempty_list_inner(comma,record_field) -> spread_record_field
 ##
 
 Expected more record fields or an immediate `}` to complete the record expression.
 
+program: LBRACE ELLIPSIS BIGINT ARROW
```

Comment:
Same thing here, next token should be a comma or a `}`.

Reply:
Hmm, should an immediate `}` be allowed here?
```
record_exprs:
  // Don't ever parse {x} as a record
  | non_punned_record_field comma? { [$1] }
  | punned_record_field comma { [$1] }
  | record_field comma lseparated_nonempty_list(comma, record_field) comma? { $1::$3 }
```
Since the `ELLIPSIS BIGINT` is the `record_field` here, the grammar would seem to indicate that only a comma must follow here.

Reply:
That's true!